File: test/java/TestStaticMethods.java
package test.java;


public class TestStaticMethods {
/*    public static boolean bool_field = true;
    public static byte byte_field = 1;
    public static char char_field = 'a';
    public static short short_field = 2;
    public static int int_field = 3;
    public static long long_field = 4;
    public static float float_field = 5.0f;
    public static double double_field = 6.0;
    public static Object object_field = new Object();
    public static String string_field = "Hello World!";
    public static int[] int_array_field = {1, 2, 3};
    public static Object[] object_array_field = {new Object(), new Object(), new Object()};*/

    public TestStaticMethods() {
        return;
    }

    public static void void_method() {
        return;
    }

    public static boolean bool_method() {
        return true;
    }

    public static byte byte_method() {
        return 1;
    }

    public static char char_method() {
        return 'a';
    }

    public static short short_method() {
        return 2;
    }

    public static int int_method() {
        return 3;
    }

    public static long long_method() {
        return 4;
    }

    public static float float_method() {
        return 5.0f;
    }

    public static double double_method() {
        return 6.0;
    }

    public static Object object_method() {
        return new TestStaticMethods();
    }

    public static String string_method() {
        return "Hello World!";
    }

    public static int[] int_array_method() {
        return new int[]{1, 2, 3};
    }

    public static Object[] object_array_method() {
        return new Object[]{new Object(), new Object(), new Object()};
    }

    public static void throws_method() throws Exception {
        throw new Exception();
    }

}

File: test/basic/class/test_static_methods.py
from unittest import TestCase

from test.utils.java import compile_java

from pyjvm.jvm import Jvm


class TestStaticMethods(TestCase):

    def get_test_class(self, jvm):
        try:
            return jvm.findClass("test/java/TestStaticMethods")
        except:
            to_load = "test/java/TestStaticMethods.java"
            class_file = compile_java(to_load)
            jvm = Jvm.aquire()

            with open(class_file, 'rb') as f:
                return jvm.loadClass(f)
            

    def test_ret_void(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.void_method()
        self.assertEqual(r, None)

    def test_ret_bool(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.bool_method()
        self.assertEqual(r, True)

    def test_ret_byte(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.byte_method()
        self.assertEqual(r, 1)
    
    def test_ret_char(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.char_method()
        self.assertEqual(r, 'a')

    def test_ret_short(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.short_method()
        self.assertEqual(r, 2)
    
    def test_ret_int(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.int_method()
        self.assertEqual(r, 3)
    
    def test_ret_long(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.long_method()
        self.assertEqual(r, 4)

    def test_ret_float(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.float_method()
        self.assertEqual(r, 5.0)
    
    def test_ret_doulbe(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.double_method()
        self.assertEqual(r, 6.0)

    def test_ret_string(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.string_method()
        self.assertEqual(r, "Hello World!")
    
    def test_ret_obj(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)
        r = test_class.object_method()
        self.assertEqual(r.__class__.__name__, "test.java.TestStaticMethods")

    def test_throws(self):
        jvm = Jvm.aquire()
        test_class = self.get_test_class(jvm)

        with self.assertRaises(Exception):
            test_class.throws_method()